botocore/client.py

```diff
@@ -554,10 +601,12 @@ class Config(object):
     def __init__(self, region_name=None, signature_version=None,
                  user_agent=None, user_agent_extra=None,
                  connect_timeout=DEFAULT_TIMEOUT,
-                 read_timeout=DEFAULT_TIMEOUT):
+                 read_timeout=DEFAULT_TIMEOUT,
+                 s3_addressing_style=None):
```

Comment:
I think we should stick with virtual and path because it is consistent with the s3 docs. I fine with ``auto`` as well. Will need to do some more testing to make sure we can support path without signing errors.